src/main/java/com/suse/saltstack/netapi/datatypes/SaltVersion.java

```diff
@@ -0,0 +1,145 @@
+package com.suse.saltstack.netapi.datatypes;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a salt version
+ */
+public class SaltVersion implements Comparable<SaltVersion> {
+
+    private static Pattern SALT_VERSION_REGEX =
+            Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)([Rr][Cc](\\d+))?$");
+
+    private final int year;
+    private final int month;
+    private final int bugfix;
+    private final Optional<Integer> releaseCandidate;
+
+    /**
+     * Parses a salt version string
+     * @param versionString the salt version string
+     * @return SaltVersion if the versionString is valid or empty Optional if not.
+     */
+    public static Optional<SaltVersion> parse(String versionString) {
```

Comment:
I think you misunderstand. I know what `Optional` does and I've written some small functional things (in Erlang, for example). My comment is that `Optional` wrapper belongs in the caller-code, not the called-code. The null checking, be that via `Optional` or otherwise, is done in caller-code anyway.

Anyway, this is all done and merged now. I just want to clarify any misunderstandings about my statements.